File: projectBlueWater/src/main/java/com/lasthopesoftware/bluewater/client/playback/engine/preparation/IPlayableFilePreparationSourceProvider.java
package com.lasthopesoftware.bluewater.client.playback.engine.preparation;

import com.lasthopesoftware.bluewater.client.playback.file.preparation.PlayableFilePreparationSource;

public interface IPlayableFilePreparationSourceProvider extends IPreparedPlaybackQueueConfiguration {
	PlayableFilePreparationSource providePlayableFilePreparationSource();
}

File: projectBlueWater/src/main/java/com/lasthopesoftware/bluewater/client/playback/engine/preparation/IPlayableFilePreparationSourceProvider.kt
package com.lasthopesoftware.bluewater.client.playback.engine.preparation

import com.lasthopesoftware.bluewater.client.playback.file.preparation.PlayableFilePreparationSource

interface IPlayableFilePreparationSourceProvider : IPreparedPlaybackQueueConfiguration {
    fun providePlayableFilePreparationSource(): PlayableFilePreparationSource
}
